Lib/asyncio/base_subprocess.py

```diff
@@ -215,9 +215,15 @@ def _process_exited(self, returncode):
             # object. On Python 3.6, it is required to avoid a ResourceWarning.
             self._proc.returncode = returncode
         self._call(self._protocol.process_exited)
-        for p in self._pipes.values():
-            if p is not None:
-                p.pipe.close()
+        # See https://github.com/python/cpython/issues/100133
+        # The pipes should not be closed otherwise some data may be lost.
+        # If the pipe is closed here then _UnixReadPipeTransport will remove the
+        # reader prematurely and the data will be lost, instead of doing that
+        # the pipe will be closed when the process is finished via _pipe_connection_lost
+        # followed by _try_finish.
+        # for p in self._pipes.values():
+        #     if p is not None:
+        #         p.pipe.close()
```

Comment:
I'd remove the commented-out code. Not sure if the whole comment is even necessary; perhaps you can shorten it?

Reply:
removed